.github/workflows/main.yml

```diff
@@ -0,0 +1,64 @@
+name: Lint and Test
+
+on: [push, pull_request]
+
+jobs:
+  list:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Python 3.8
+        uses: actions/setup-python@v2
+        id: setup-python
+        with:
+          python-version: 3.8
+      - name: Setup Pipenv
+        uses: dschep/install-pipenv-action@v1
+      - name: Install dependecies
+        run: pipenv install --dev --skip-lock --python ${{ steps.setup-python.outputs.python-version }}
+      - name: Lint code
+        run: pipenv run lint --check
+      - name: Lint writing
+        uses: actionshub/markdownlint@main
+
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.6, 3.9]
+        include:
+          - os: macos-latest
+            python-version: 3.8
+          - os: windows-latest
+            python-version: 3.7
```

Comment:
This matrix is what I came up with to both keep usage and time down, while providing the best coverage. @schultzh @gkapfham what do you think about this? Should we try testing all combinations instead, or some other subset?

Reply:
Hi @Michionlion I understand why you are suggesting this approach and it is fine with me. Although I don't think that we should do the experiment before merging this PR, it would be interesting to find out how much time we are likely to save with this approach in comparison to the naive one that runs all of the versions of Python on all of the operating systems.

Finally, I'm glad to see that we are testing on Python 3.9 as there are currently a number of Pipenv-related bugs that I'm tracking that are currently unresolved only for Python 3.9!

(Side note: I'd really like to move over to Poetry instead of Pipenv for GatorGrader. I've adopted it for all of my other major Python projects, I use it in all of my teaching now, and, overall, I like it much better than Pipenv!)